src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -211,9 +218,19 @@ function ReportActionItemMessageEdit(props) {
     const deleteDraft = useCallback(() => {
         debouncedSaveDraft.cancel();
         Report.saveReportActionDraft(props.reportID, props.action.reportActionID, '');
-        ComposerActions.setShouldShowComposeInput(true);
-        ReportActionComposeFocusManager.clear();
-        ReportActionComposeFocusManager.focus();
+
+        if (isFocusedRef.current || EmojiPickerAction.isActive(props.action.reportActionID) || ReportActionContextMenu.isActiveReportAction(props.action.reportActionID)) {
+            setShouldShowComposeInputKeyboardAware(true);
+            isFocusedRef.current = true;
```

Comment:
We have discussed the new issue of re-focusing for a brief moment [here](https://expensify.slack.com/archives/C049HHMV9SM/p1692747837798019). This issue is introduced when I add `onMouseDown={e => e.preventDefault()}` props to the `save` button for prevent loosing focus.

`deleteDraft` method is called right before unmount is called. Setting this value ensures that the current action is active. We can fix the above issue(both Enter key or Save button) by this change

Reply:
I removed the above code. In order to prevent auto refocusing of composer input after delete modal is closed, I blurred the input element before showing the confirm modal
Please check the latest code

Reply:
We want to prevent refocusing on that input if we press Delete on the delete modal right? But what if we press Close on the modal? Will the input get refocused again? (check on both web and native)

Reply:
@s77rt 

This issue already exists on staging and not a regression. Will you check again?

Reply:
Yeah not something that we should be concerned with here. Safari limit the focus functionality.